charts/k8s-monitoring/templates/agent_config/_processors.river.txt

```diff
@@ -47,11 +47,25 @@ otelcol.processor.k8sattributes "default" {
     logs    = [otelcol.processor.transform.add_resource_attributes.input]
 {{- end }}
 {{- if .Values.traces.enabled }}
-    traces  = [otelcol.processor.transform.add_resource_attributes.input]
+    traces  = [
+      otelcol.processor.transform.add_resource_attributes.input, 
+    {{- if .Values.metrics.enabled }}
+      otelcol.connector.host_info.default.input,
+    {{- end }}
+    ]
 {{- end }}
   }
 }
 
+{{- if and .Values.metrics.enabled .Values.traces.enabled }}
+otelcol.connector.host_info "default" {
+  host_identifiers = [ "k8s.node.name" ]
+  output {
+    metrics = [otelcol.processor.batch.batch_processor.input]
+  }
+}
+{{- end }}
```

Comment:
This component should live closer to remote write. Something like this should allow us to skip the extra metric processing for the host info metrics:
```suggestion
    metrics = [otelcol.processor.batch.host_info_batch.input]
  }
}

otelcol.processor.batch "host_info_batch" {
  metrics = [otelcol.exporter.prometheus.host_info_metrics.input]
}

otelcol.exporter.prometheus "host_info_metrics" {
  add_metric_suffixes = false
  forward_to          = [prometheus.remote_write.metrics_service.receiver]
}

{{- end }}
```